src/infobar.rs

```diff
@@ -6,29 +6,44 @@ use crate::paths::Paths;
 
 /// Text contains the strings required to print the infobar.
 pub struct Text {
-    /// current image is the string of the current image path
-    pub current_image: String,
-    /// index is the string represention of the index.
-    pub index: String,
+    /// Either displays the name of the current image or the current command the user is typing in
+    /// command mode
+    pub information: String,
+    /// In normal mode this is the string represention of the index, in command mode this is
+    /// "Command"
+    pub mode: String,
 }
 
-impl From<&Paths> for Text {
-    fn from(p: &Paths) -> Self {
-        let current_image = match p.images.get(p.index) {
-            Some(path) => match path.to_str() {
-                Some(name) => name.to_string(),
-                None => "No file".to_string(),
-            },
-            None => "No file selected".to_string(),
-        };
-        let index = if p.images.is_empty() {
-            "No files in path".to_string()
-        } else {
-            format!("{} of {}", p.index + 1, p.max_viewable)
-        };
-        Text {
-            current_image,
-            index,
+impl Text {
+    /// Updates the infobar
+    /// if cmd isn't empty the user is in command mode and therefore that should be displayed
+    /// instead of normal mode information. Normal mode information is the index and the current
+    /// image path
+    pub fn update(paths: &Paths, msg: Option<&str>) -> Self {
+        let information: String;
+        let mode: String;
+        match msg {
+            Some(msg) => {
+                // user is in command mode
+                information = msg.to_string();
+                mode = String::from("Command");
```

Comment:
Am I right in thinking that if I send Some() string to this function, then I'll end up in Command mode? Will this effect using the infobar for error messages?

Reply:
Yeah, this was the main issue addressed in "Rewrite of Modal system".  New changes
```rust
pub enum Mode {
    /// Default mode, allows the removal, traversal, move, and copy of images
    Normal,
    /// Mode that is built off of user input, allows switching the current glob
    /// string is the input to display on the infobar
    Command(String),
    /// Mode that is meant to display errors to the user through the infobar
    /// string is the input to display on the infobar
    Error(String),
    /// Terminate condition, if this mode is set the program will stop execution
    Exit,
}
```
The `render_infobar` funciton now passes in the current mode, which contains the message if applicable and This required me to change a lot of the way I handled error handling, so now Error was added as an easy solution.